chalk-engine/src/logic.rs

```diff
@@ -126,6 +126,44 @@ impl<C: Context> Forest<C> {
             .any(|strand| test(&strand.ex_clause.subst))
     }
 
+    fn ensure_precheck(
```

Comment:
Needs a comment and a better name, I think. "precheck" doesn't give me a lot of feeling for what it does.

Reply:
Yep, I agree :) Name was changed in later commits.